pkg/cloudprovider/cloudprovider.go

```diff
@@ -270,12 +313,29 @@ func (c *CloudProvider) waitUntilLaunched(ctx context.Context, nodeClaim *karpv1
 }
 
 func (c *CloudProvider) List(ctx context.Context) ([]*karpv1.NodeClaim, error) {
+	var nodeClaims []*karpv1.NodeClaim
+
+	// List AKS machine-based nodes
```

Comment:
minor, style, you can ignore if you want: It might help this read a bit cleaner if instead of comment-block sections you just put each of the machine vs vm sections into their own helper methods:

```
machineClaims, err := c.listMachines(...)
...

vmClaims, err := c.listVMs(...)
...

return lo.Concat(machineClaims, vmClaims)
```